nf_core/components/install.py

```diff
@@ -0,0 +1,266 @@
+import logging
+import os
+import re
+from pathlib import Path
+
+import questionary
+
+import nf_core.modules.modules_utils
+import nf_core.utils
+from nf_core.components.components_command import ComponentCommand
+from nf_core.components.components_utils import prompt_component_version_sha
+from nf_core.modules.modules_json import ModulesJson
+from nf_core.modules.modules_repo import NF_CORE_MODULES_NAME
+
+log = logging.getLogger(__name__)
+
+
+class ComponentInstall(ComponentCommand):
+    def __init__(
+        self,
+        pipeline_dir,
+        component_type,
+        force=False,
+        prompt=False,
+        sha=None,
+        remote_url=None,
+        branch=None,
+        no_pull=False,
+        installed_by=False,
+    ):
+        super().__init__(component_type, pipeline_dir, remote_url, branch, no_pull)
+        self.force = force
+        self.prompt = prompt
+        self.sha = sha
+        if installed_by:
+            self.installed_by = installed_by
+        else:
+            self.installed_by = self.component_type
+
+    def install(self, component, silent=False):
+        if self.repo_type == "modules":
+            log.error(f"You cannot install a {component} in a clone of nf-core/modules")
+            return False
+        # Check whether pipelines is valid
+        if not self.has_valid_directory():
+            return False
+
+        # Check modules directory structure
+        self.check_modules_structure()
+
+        # Verify that 'modules.json' is consistent with the installed modules and subworkflows
+        modules_json = ModulesJson(self.dir)
+        modules_json.check_up_to_date()
+
+        # Verify SHA
+        if not self.modules_repo.verify_sha(self.prompt, self.sha):
+            return False
+
+        # Check and verify component name
+        component = self.collect_and_verify_name(component, self.modules_repo)
+        if not component:
+            return False
+
+        # Get current version
+        current_version = modules_json.get_component_version(
+            self.component_type, component, self.modules_repo.remote_url, self.modules_repo.repo_path
+        )
+
+        # Set the install folder based on the repository name
+        install_folder = os.path.join(self.dir, self.component_type, self.modules_repo.repo_path)
+
+        # Compute the component directory
+        component_dir = os.path.join(install_folder, component)
+
+        # Check that the component is not already installed
+        component_installed = self.check_component_installed(
+            component, current_version, component_dir, self.modules_repo, self.force, self.prompt, silent
+        )
+        if not component_installed:
+            log.debug(
+                f"{self.component_type[:-1].title()} is already installed and force is not set.\nAdding the new installation source {self.installed_by} for {self.component_type[:-1]} {component} to 'modules.json' without installing the {self.component_type}."
```

Comment:
doesn't this break installing new components? Because they will have `component_installed=Falsed` and will then be kicked out of the install function at this `return False` statement, no?

Reply:
It was actually the other way around, the function returned true when the module was not installed. I modified the True/False logic to make it more understandable